Source/app/routes.js

```diff
@@ -168,14 +168,41 @@ module.exports = function(app, passport) {
     app.post('/addBuilding', function (req, res) {
 
        
-        DB.addBuildingDatabase(req.body);
+        addBuildingDatabase(req.body);
         res.send(req.body);
 
     });
 
 }
-
-
+function addEntryToDatabase(entry) {
+    console.log('Inside addEntryToDatabase')
+    /*
+    var data = new DataEntry();
+    data.point = entry.das.devices.device.records.record.point;
+    data.timestamp = entry.das.devices.device.records.record.time._;
+    res 
+    */
+};  
+function addBuildingToDatabase(entry) {
+    Building.findOne({name: entry.building.name}, function (err, docs) {
+          if(docs === null){ // ensure building doesn't exist
+              var build = new Building();
+               // set all of the relevant information
+              build.name = entry.building.name
+              build.building_type = entry.building.building_type;
+              // serial can be used as identifier when adding data (data has serial # of AcquiSuite)
+              build.serial = entry.building.serial;
+              // save the building
+              build.save()
+                   .catch( err => {res.status(400)
+                   .send("unable to save to database");})
+              console.log("The building '" + entry.building.name + "' has been added.");
+          }
+          else
+              console.log('Nothing was added');     
```

Comment:
I believe this will cause async errors, where the mongoose .save function will not finish before the route handler tries to res.send data and will send a null value.

The res.send needs to be in your ,save function to ensure order of steps.

see examples already implemented.

Have you tested that this works?

Reply:
I think the findONE and mongoos calls need to be IN the route handler

Reply:
Yes this works as it is.
It won't have async issues because it's calls the addBuildingToDatabase after data has been parsed asynchronously